test/extended/router/metrics.go

```diff
@@ -186,7 +186,73 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 			o.Expect(findGaugesWithLabels(metrics["haproxy_backend_up"], routeLabels)).To(o.Equal([]float64{1}))
 			o.Expect(findGaugesWithLabels(metrics["haproxy_server_bytes_in_total"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">=", 0), o.BeNumerically(">=", 0)))
 			o.Expect(findGaugesWithLabels(metrics["haproxy_server_bytes_out_total"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">=", 0), o.BeNumerically(">=", 0)))
-			o.Expect(findGaugesWithLabels(metrics["haproxy_server_max_sessions"], serverLabels)).To(o.ConsistOf(o.BeNumerically(">", 0), o.BeNumerically(">", 0)))
+
+			// haxproxy_server_max_sessions needs to be
+			// greater than zero over two successive
+			// scrapes once the first scrape shows an
+			// increase in total connections.
+			err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+				refreshMetrics := func() (map[string]*dto.MetricFamily, error) {
+					startTime := time.Now()
+					results, err := getBearerTokenURLViaPod(ns, execPodName, fmt.Sprintf("http://%s:%d/metrics", host, metricsPort), bearerToken)
+					if err != nil {
+						return nil, err
+					}
+					metrics, err := p.TextToMetricFamilies(bytes.NewBufferString(results))
+					e2e.Logf("metrics refresh completed in %s", time.Now().Sub(startTime).Round(time.Second).String())
+					return metrics, err
+				}
+
+				maxSessionsInvariant := func(v []float64) bool {
+					return len(v) >= 2 && v[0] > 0 && v[1] > 0
```

Comment:
Are there situations in this test where the `maxSessionsInvariant` function should return true for slices with more than 2 values?

Reply:
Not any more as the test got moved to the end of the suite and there we only have 1 endpoint now.